core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -269,6 +269,23 @@ private[spark] object Utils extends Logging {
     file.setExecutable(true, true)
   }
 
+  def moveToTrashIfEnabled(
+      fs: FileSystem,
+      partitionPath: Path,
+      trashInterval: Int,
+      hadoopConf: Configuration): Unit = {
+    if (trashInterval < 0) {
+      if (!fs.delete(partitionPath, true)) {
+        throw new RuntimeException(
```

Comment:
Why do you want to change the existing behavior?

Reply:
To avoid code duplication i used `moveToTrashIfEnabled` function to handle both `truncate table` and `insert overwrite` This exception is thrown in case of insert overwrite in `sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala`

If you suggest, i can modify and handle this separately.

Reply:
Do you think this new code is correct in `TruncateTableCommand`?

Reply:
Hi @dongjoon-hyun, I have corrected and kept the old behavior. Kindly check.